### PR TITLE
rec: Count DNSSEC stats for some names in a different set of counters

### PR DESCRIPTION
**Pull request by omoerbeek**
### Short description

This implements the feature as requested in #10058.

Idea is that you can set `x-dnssec-names` to a suffixmatch list. DNSSEC counting actions will then be done for on `x-dnssec-result-bogus-...` counter instead of the regular one.

Missing: 
- [ ] Adding suffixes using `rec_control`.
- [x] Marking of log lines as requested in the feature request.
- [x] Docs (settings + metrics)

This PR is mainly to gauge if this approach is the way forward. I'm also open for a better name than `x-dnssec...`

### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [X] compiled this code
- [X] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

**Comment by omoerbeek** [comment]
Note that I count in one set or another, not both as suggested by the feature request.
Also, I might consider adding the `x-` metrics only if the corresponding setting is non-empty.

**Comment by omoerbeek** [comment]
We decidd to leave the `rec_control` manipulation of `x-dnssec-names` out for now.

File: pdns/pdns_recursor.cc [hunk 1]
```diff
@@ -1835,26 +1837,34 @@ static void startDoResolve(void *p)
...
+          if (sr.doLog() || vStateIsBogus(state)) {
+            auto xdnssec = g_xdnssec.getLocal();
+            if (xdnssec->check(dc->d_mdp.d_qname)) {
+              x_marker = " [in x-dnssec-names]";
+            }
           }
 
-          auto state = sr.getValidationState();
+          if(sr.doLog()) {
+            g_log<<Logger::Warning<<"Starting validation of answer to "<<dc->d_mdp.d_qname<<"|"<<QType(dc->d_mdp.d_qtype).getName()<<x_marker<<" for "<<dc->getRemote()<<endl;
```
Comment (rgacogne) [inline pdns/pdns_recursor.cc:1851]: I think this line should go, actually. It used to make sense in 4.0 when we were doing validation here, after everything else, but I think it does not make sense anymore.

**Review by rgacogne — APPROVED** [review]
Looks good to me! I wonder if we will actually use the separate metrics, or if we could get away with not accounting the validation states for the names in that list at all. On the other hand we have often been very happy to have additional metrics when investigating an issue, and in that case it could for example make it clear that a mistake has been made by adding the wrong name to the "separate" list, for example.

**Pull request merged**